shared/react-native/ios/Keybase/AppDelegate.m

```diff
@@ -87,6 +87,60 @@ - (void) createBackgroundReadableDirectory:(NSString*) path setAllFiles:(BOOL)se
   }
 }
 
+- (BOOL) maybeMigrateDirectory:(NSString*)source dest:(NSString*)dest {
+  NSError* error = nil;
+  NSFileManager* fm = [NSFileManager defaultManager];
+
+  // Always do this copy in case it doesn't work on previous attempts. 
+  NSArray<NSString*>* sourceContents = [fm contentsOfDirectoryAtPath:source error:&error];
+  if (nil == sourceContents) {
+    NSLog(@"Error listing app contents directory: %@", error);
+    return NO;
+  } else {
+    for (NSString* file in sourceContents) {
+      BOOL isDirectory = NO;
+      NSString* path = [NSString stringWithFormat:@"%@/%@", source, file];
+      NSString* destPath = [NSString stringWithFormat:@"%@/%@", dest, file];
+      if ([fm fileExistsAtPath:path isDirectory:&isDirectory] && isDirectory) {
+        NSLog(@"skipping directory: %@", file);
+        continue;
+      }
+      if (![fm copyItemAtPath:path toPath:destPath error:&error]) {
+        if ([error code] == NSFileWriteFileExistsError) {
+          // Just charge forward if the file is there already
+          continue;
+        }
+        NSLog(@"Error copying file: %@ error: %@", file, error);
+        return NO;
+      }
+    }
+  }
+  return YES;
+}
+
+- (NSString*) setupHome:(NSString*)home sharedHome:(NSString*)sharedHome {
+  // Setup all directories
+  NSString* appKeybasePath = [@"~/Library/Application Support/Keybase" stringByExpandingTildeInPath];
+  NSString* appEraseableKVPath = [@"~/Library/Application Support/Keybase/eraseablekvstore/device-eks" stringByExpandingTildeInPath];
+  NSString* sharedKeybasePath = [NSString stringWithFormat:@"%@/Library/Application Support/Keybase", sharedHome];
+  [self createBackgroundReadableDirectory:appKeybasePath setAllFiles:YES];
+  [self createBackgroundReadableDirectory:sharedKeybasePath setAllFiles:YES];
+  [self createBackgroundReadableDirectory:appEraseableKVPath setAllFiles:YES];
+  [self addSkipBackupAttributeToItemAtPath:appKeybasePath];
+  [self addSkipBackupAttributeToItemAtPath:sharedKeybasePath];
+  
+  if (![self maybeMigrateDirectory:appKeybasePath dest:sharedKeybasePath]) {
+    return home;
+  }
+  NSString* sharedEraseableKVPath =  [NSString stringWithFormat:@"%@/Library/Application Support/Keybase/eraseablekvstore/device-eks", sharedHome];
```

Comment:
why not move this up to the other path definitions?

Reply:
I originally had a check to see if the shared dir was empty before running this copy, and so this directory was breaking that. I removed it since I realized that if the app is killed during this migration it would hose your install.